test/blackbox/common/RTPSBlackboxTestsBasic.cpp

```diff
@@ -866,6 +866,35 @@ TEST(RTPS, MultithreadedWriterCreation)
     RTPSDomain::stopAll();
 }
 
+/* Regression Test for improving gaps processing
+ *  https://github.com/eProsima/Fast-DDS/pull/3343
+ */
+TEST(RTPS, RTPSCorrectGAPProcessing)
+{
+    RTPSWithRegistrationReader<HelloWorldPubSubType> reader(TEST_TOPIC_NAME);
+    RTPSWithRegistrationWriter<HelloWorldPubSubType> writer(TEST_TOPIC_NAME);
```

Comment:
I don't think we need a writer here

Reply:
The thing is that we need a matched writer otherwise `processGAP()` skips in `acceptMsgFrom()` and the `assert` would not be evaluated .

Other option would be setting `m_acceptMessagesFromUnkownWriters = true` but I dont think it can be changed from outside.

Reply:
You are right. I overlooked that bit.